### PR TITLE
feature/BB 358/supportNewerNoncurrentVersion

### PR DESCRIPTION
**Pull request by alexanderchan-scality**
- BB-358: add support for NewerNonccurrentVersions
- BB-358: add tests for NewerNoncurrentVersion helpers
- bump aws-sdk to v2.1035.0

### CODE REVIEW

File: extensions/lifecycle/tasks/LifecycleTask.js [hunk 1]
```diff
@@ -241,7 +374,23 @@ class LifecycleTask extends BackbeatTask {
...
+                    if (err) {
+                        return done(err);
+                    }
+
+                    if (data.IsTruncated) {
+                        this._ncvHeapBucketClear(bucketData.target.bucket);
+                        return done();
+                    }
+
+                    this._ncvHeapObjectsClear(
```
Comment (nicolas2bert) [inline extensions/lifecycle/tasks/LifecycleTask.js]: Should we not keep NextMarker versions only if the listing is truncated?
Also, you might want to add a comment to ease the code review.

Is this pagination logic tested somewhere?

Comment (alexanderchan-scality) [reply extensions/lifecycle/tasks/LifecycleTask.js]: with the uniqueKey set, NextMarkers key versions are kept when IsTruncated is true. 

you're correct. the logic is flipped here ; the  IsTruncated check should be for falsy as the clearing of the buckets should only be done at the end of the listing.

**Review by nicolas2bert — APPROVED** [review]

**Review by naren-rajendran — APPROVED** [review]

**Pull request merged**